RAID/refresh_secondary

```diff
@@ -67,170 +69,184 @@ if ! mdadm --detail /dev/md0 2>&1 | grep -q ", degraded $" ; then
 fi
 
 # Perform sanity checks on primary disk first
-if ! mdadm --detail /dev/md0 | grep -q /dev/sda; then
-	echo "ERROR: \"Primary\" disk /dev/sda is not part of RAID /dev/md0!!"
+SDA=$(ls -l /sys/block/sd* | cut -d'>' -f2 | grep /ata | sort | head -n 1)
+SDA=$(basename $SDA)
+
+if ! mdadm --detail /dev/md0 | grep -q /dev/$SDA; then
+	echo "ERROR: \"Primary\" disk /dev/$SDA is not part of RAID /dev/md0!!"
 	exit 1
 fi
-if [ "$(cat /sys/block/sda/removable)" != "0" ]; then
-	echo "WARNING: \"Primary\" disk /dev/sda is removable:"
-	echo -n " /dev/sda: "$(lsblk --nodeps -no tran /dev/sda | tr [:lower:] [:upper:]) 
-	echo -n " "$(lsblk --nodeps -no type /dev/sda)
-	echo -n ", model" $(lsblk --nodeps -no model /dev/sda)
-	echo ", s/n "$(lsblk --nodeps -no serial /dev/sda)
+if [ "$(cat /sys/block/"$SDA"/removable)" != "0" ]; then
+	echo "WARNING: \"Primary\" disk /dev/$SDA is removable:"
+	echo -n " /dev/"$SDA": "$(lsblk --nodeps -no tran /dev/$SDA | tr [:lower:] [:upper:]) 
+	echo -n " "$(lsblk --nodeps -no type /dev/$SDA)
+	echo -n ", model" $(lsblk --nodeps -no model /dev/$SDA)
+	echo ", s/n "$(lsblk --nodeps -no serial /dev/$SDA)
 	echo -n "Press <Enter> if you wish to continue anyway (Control-C aborts) "
 	read $ans
 fi
-if $(partprobe -ds /dev/sda | grep -q msdos) && [ ! -d /boot/efi ]; then
+if $(partprobe -ds /dev/$SDA | grep -q msdos) && [ ! -d /boot/efi ]; then
 	boot="MBR"
-elif [ ! -z "$(sgdisk -i1 /dev/sda | grep 'BIOS boot')" ] && [ ! -d /boot/efi ]; then
+elif [ ! -z "$(sgdisk -i1 /dev/$SDA | grep 'BIOS boot')" ] && [ ! -d /boot/efi ]; then
 	boot="BIOS";
-elif [ ! -z "$(sgdisk -i1 /dev/sda | grep 'EFI [Ss]ystem')" ] && [ -d /boot/efi ]; then
+elif [ ! -z "$(sgdisk -i1 /dev/$SDA | grep 'EFI [Ss]ystem')" ] && [ -d /boot/efi ]; then
 	boot="EFI";
 else
-	echo "ERROR: Unrecognised boot scheme on \"Primary\" disk /dev/sda"
+	echo "ERROR: Unrecognised boot scheme on \"Primary\" disk /dev/$SDA"
 	exit 1
 fi
 
 # Now get the user to load the Secondary disk
-if [ -b $disk ] && [ $ACCEPT_LOAD == 0 ] && ([ ! -O $lockf ] || [ "$(cat $lockf)" != "$(lsblk --nodeps -no serial $disk)" ]); then
-	echo "ERROR:\"Secondary\" disk $disk must not be loaded before start of script"
+SDB=$(ls -l /sys/block/sd* | cut -d'>' -f2 | grep /ata | sort | head -n 2 | tail -n 1)
+SDB=$(basename $SDB)
+
+if [ $SDB != $SDA ] && [ $ACCEPT_LOAD == 0 ] && ([ ! -O $lockf ] || [ "$(cat $lockf)" != "$(lsblk --nodeps -no serial /dev/$SDB)" ]); then
+	echo "ERROR:\"Secondary\" disk /dev/$SDB must not be loaded before start of script"
 	echo "(You may want to use the '-A' option to [A]llow if this is intended)"
 	exit 1
 fi
-if [ ! -b $disk ]; then
-	while [ ! -b $disk ]; do
-		echo  -n -e "\rWaiting for disk $disk to be loaded, use 'Control-C' to safely abort ... "
+if [ $SDA == $SDB ] || [ ! -b /dev/$SDB ]; then
+	while [ $SDA == $SDB ] || [ ! -b /dev/$SDB ]; do
```

Comment:
I am curious why the condition with `-b` is needed. Maybe it is belt-and-suspenders.

Reply:
Belt and braces indeed.  It also defeats any possible race condition with udev's action if the device and its symlinks should not be instanced simultaneously.